### PR TITLE
irmin-pack: disable the layered store

### PR DESCRIPTION
**Pull request by CraigFe**
The implementation is currently incompatible with direct pointers. Support for the layered store will be restored after direct pointers has been merged.

### CODE REVIEW

File: bench/irmin-pack/trace_definitions.ml [hunk 1]
```diff
@@ -317,7 +317,7 @@ module Stat_trace = struct
 
     type config = {
       inode_config : int * int * int;
-      store_type : [ `Pack | `Pack_layered | `Pack_mem ];
+      store_type : [ `Pack | `Pack_mem ];
```
Comment (Ngoguey42) [inline bench/irmin-pack/trace_definitions.ml:320]: This diff should be reverted in order to keep preserve the file format

File: bench/irmin-pack/trace_stat_summary_pp.ml [hunk 2]
```diff
@@ -147,7 +147,6 @@ module Table0 = struct
     | `Store_type -> (
         match s.config.store_type with
         | `Pack -> "pack"
-        | `Pack_layered -> "pack-layered"
```
Comment (Ngoguey42) [inline bench/irmin-pack/trace_stat_summary_pp.ml:150]: This diff as well

File: bench/irmin-pack/trace_definitions.ml (hunk 1 above)
Comment (CraigFe) [reply bench/irmin-pack/trace_definitions.ml]: Done, thanks.

**Review by icristescu — APPROVED** [review]
Thanks, LGTM!

**Comment by CraigFe** [comment]
@Ngoguey42, can you take another look to make sure I've fixed the replay format regression?

**Comment by Ngoguey42** [comment]
Looks good

**Pull request merged**